addons/hr/models/res_partner.py

```diff
@@ -62,3 +63,40 @@ def _compute_employee(self):
         employees = {employee for [employee] in employee_data}
         for partner in self:
             partner.employee = partner in employees
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_contact_rel_employee(self):
+        partners = self.filtered(lambda partner: partner.employee)
+        if len(self) == 1 and len(partners) == 1 and self.id == partners[0].id:
+            raise UserError(_('You cannot delete contact that are linked to an employee, please archive them instead.'))
+        if len(partners) > 0:
```

Comment:
it can be like this only and it will be false if it is empty record-set 
```suggestion
        if partners:
```